src/main/java/ai/shreds/adapter/AdapterSupplierController.java

```diff
@@ -0,0 +1,63 @@
+package ai.shreds.adapter;
+
+import ai.shreds.application.ApplicationGetAllSuppliersInputPort;
+import ai.shreds.application.ApplicationGetSupplierByIdInputPort;
+import ai.shreds.shared.SharedRequestParams;
+import ai.shreds.shared.SharedSupplierDTO;
+import ai.shreds.adapter.AdapterSupplierMapper;
+import ai.shreds.adapter.AdapterSupplierResponseDTO;
+import ai.shreds.domain.SupplierNotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/suppliers")
+@RequiredArgsConstructor
+public class AdapterSupplierController {
+
+    private final ApplicationGetAllSuppliersInputPort getAllSuppliersInputPort;
+    private final ApplicationGetSupplierByIdInputPort getSupplierByIdInputPort;
+    private final AdapterSupplierMapper supplierMapper;
+
+    @GetMapping
+    public ResponseEntity<List<AdapterSupplierResponseDTO>> getAllSuppliers(AdapterSupplierRequestParams params) {
+        SharedRequestParams sharedParams = new SharedRequestParams(params.getName(), params.getContact_info(), params.getAddress());
+        List<SharedSupplierDTO> suppliers = getAllSuppliersInputPort.getAllSuppliers(sharedParams);
+        List<AdapterSupplierResponseDTO> response = suppliers.stream()
+                .map(supplierMapper::toDTO)
```

Comment:
change this line:
.map(supplierMapper::toDTO)
to:
.map(supplierMapper::toAdapterSupplierResponseDTO)

add the toAdapterSupplierResponseDTO method in the AdapterSupplierMapper class